pkg/scalers/azure/azure_data_explorer.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	"github.com/Azure/azure-kusto-go/kusto/data/table"
+	"github.com/Azure/azure-kusto-go/kusto/unsafe"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DataExplorerMetadata struct {
+	ClientID     string
+	ClientSecret string
+	DatabaseName string
+	Endpoint     string
+	MetricName   string
+	PodIdentity  string
+	Query        string
+	TenantID     string
+	Threshold    int64
+}
+
+var azureDataExplorerLogger = logf.Log.WithName("azure_data_explorer_scaler")
+
+func CreateAzureDataExplorerClient(metadata *DataExplorerMetadata) (*kusto.Client, error) {
+	authConfig, err := getDataExplorerAuthConfig(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get data explorer auth config: %v", err)
+	}
+
+	client, err := kusto.New(metadata.Endpoint, kusto.Authorization{Config: *authConfig})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kusto client: %v", err)
+	}
+
+	return client, nil
+}
+
+func getDataExplorerAuthConfig(metadata *DataExplorerMetadata) (*auth.AuthorizerConfig, error) {
+	var authConfig auth.AuthorizerConfig
+
+	if metadata.PodIdentity != "" {
+		authConfig = auth.NewMSIConfig()
+		azureDataExplorerLogger.Info("Creating Azure Data Explorer Client using Pod Identity")
```

Comment:
We might want to print these messages on DEBUG level, to not flood log with messages.
Could you please convert all `.Info()` messages in this file to `.V(1).Info()`?

Reply:
done